tests/sqlparser_bigquery.rs

```diff
@@ -2097,6 +2097,14 @@ fn test_bigquery_create_function() {
             "REMOTE WITH CONNECTION us.myconnection ",
             "OPTIONS(a = [1, 2])",
         ),
+        // Templated
+        concat!(
+            "CREATE OR REPLACE TEMPORARY FUNCTION ",
+            "my_function(param1 ANY TYPE) ",
+            "AS (",
+            "(SELECT 1)",
+            ")",
+        ),
```

Comment:
Since the functionality itself is the data type, I'm thinking we can use a dedicated test for it? e.g `fn test_parse_any_type()`? Also can we add a test case to demonstrate the behavior in the other comment what happens in a sql like `CREATE TABLE foo (x ANY);`

Reply:
Good point, I was in fact doing that but reverted to inlining the test with other UDFs. Not being sure what was the preferred pattern but makes sense!